crates/ruff_linter/src/rules/perflint/rules/manual_list_comprehension.rs

```diff
@@ -406,7 +406,18 @@ fn convert_to_list_extend(
     };
     let target_str = locator.slice(for_stmt.target.range());
     let elt_str = locator.slice(to_append);
-    let generator_str = format!("{elt_str} {for_type} {target_str} in {for_iter_str}{if_str}");
+    let formatted_elt = if to_append.is_generator_expr() {
+        if elt_str.starts_with('(') && elt_str.ends_with(')') {
+            elt_str.to_string()
+        } else {
+            format!("({elt_str})")
+        }
+    } else {
+        elt_str.to_string()
+    };
+
+    let generator_str =
+        format!("{formatted_elt} {for_type} {target_str} in {for_iter_str}{if_str}");
```

Comment:
I forgot that generators have a `parenthesized` field already. What do you think about this?

```suggestion
    let generator_str = if to_append
        .as_generator_expr()
        .is_some_and(|generator| !generator.parenthesized)
    {
        format!("({elt_str}) {for_type} {target_str} in {for_iter_str}{if_str}")
    } else {
        format!("{elt_str} {for_type} {target_str} in {for_iter_str}{if_str}")
    };
```